service/raft/group0_state_machine.cc

```diff
@@ -393,6 +436,12 @@ future<> group0_state_machine::apply(std::vector<raft::command_cref> command) {
         auto is = ser::as_input_stream(c);
         auto cmd = ser::deserialize(is, boost::type<group0_command>{});
 
+#ifndef SCYLLA_BUILD_MODE_RELEASE
```

Comment:
We should avoiding the usage of macros directly in code like this.

We can leave it like this for now, but as a follow-up, once @bitpathfinder's PR which adds the nice constexpr values gets merged, we can replace this macro usage with a constexpr if.

Reply:
> We should avoiding the usage of macros directly in code like this.

This is not a macro. but a simple guard. Where the rule that it cannot be used comes from suddenly? In fact it is used in the code.

On the other topic why do it in dev mode? Dev is not suppose to be slow. It suppose to compile fast. Lets leave slow checks for debug.

Reply:
> On the other topic why do it in dev mode? Dev is not suppose to be slow. It suppose to compile fast. Lets leave slow checks for debug.

This check is not that slow, and it happens in group 0 apply which is not a hot path. Remember we don't run all tests in debug mode. Better to leave the check in dev. It won't have any observable significant effect.

We also have error injections in dev.

Reply:
> > On the other topic why do it in dev mode? Dev is not suppose to be slow. It suppose to compile fast. Lets leave slow checks for debug.
> 
> This check is not that slow, and it happens in group 0 apply which is not a hot path. Remember we don't run all tests in debug mode. Better to leave the check in dev. It won't have any observable significant effect.
> 
> We also have error injections in dev.

Error injection adds an `if` if not enabled. Here we have a loop on each apply (btw what about snapshot?). We have very limited number of tests that do not run in dev mode.

Reply:
Yeah, not really sure about this one in this case ... I'd personally think it might be good to keep the check also in dev, because when someone adds a new group0 table but not the group0 flag (and is only using dev for testing locally), they would only see the failure once pushed and failing in the CI debug. I'd think it might be better to see the failure already when testing locally.